quartz/plugins/transformers/syntax.ts

```diff
@@ -1,20 +1,41 @@
 import { QuartzTransformerPlugin } from "../types"
 import rehypePrettyCode, { Options as CodeOptions } from "rehype-pretty-code"
+import { BuiltinTheme } from "shikiji"
 
-export const SyntaxHighlighting: QuartzTransformerPlugin = () => ({
-  name: "SyntaxHighlighting",
-  htmlPlugins() {
-    return [
-      [
-        rehypePrettyCode,
-        {
-          keepBackground: false,
-          theme: {
-            dark: "github-dark",
-            light: "github-light",
-          },
-        } satisfies Partial<CodeOptions>,
-      ],
-    ]
+interface Theme {
+  light?: BuiltinTheme
+  dark?: BuiltinTheme
+}
+
+interface Options {
+  theme?: Theme
+  keepBackground?: boolean
+}
+
+const defaultOptions: Options = {
+  theme: {
+    light: "github-light",
+    dark: "github-dark",
   },
-})
+  keepBackground: false,
+}
+
+export const SyntaxHighlighting: QuartzTransformerPlugin<Options> = (opts?: Partial<Options>) => {
+  const { theme = defaultOptions.theme || {}, keepBackground } = { ...defaultOptions, ...opts }
+  const { light = defaultOptions.theme?.light, dark = defaultOptions.theme?.dark } = theme
+
+  return {
+    name: "SyntaxHighlighting",
+    htmlPlugins() {
+      return [
+        [
+          rehypePrettyCode,
+          {
+            theme: { light, dark },
+            keepBackground,
+          } as Partial<CodeOptions>,
```

Comment:
```suggestion
export const SyntaxHighlighting: QuartzTransformerPlugin<Options> = (userOpts?: Partial<Options>) => {
  const opts = { ...defaultOptions, ...userOpts }

  return {
    name: "SyntaxHighlighting",
    htmlPlugins() {
      return [
        [
          rehypePrettyCode,
					opts as Partial<CodeOptions>,
```

Reply:
That was silly of me, now that I look back on it. Thank you kindly for your review, Aaron!